js/ui/appDisplay.js

```diff
@@ -2818,8 +2858,15 @@ class HackAppIcon extends AppIcon {
         let iconFile = Gio.File.new_for_uri(iconUri);
         let gicon = new Gio.FileIcon({ file: iconFile });
 
-        return new St.Icon({ gicon: gicon,
-                             icon_size: iconSize });
+        let icon = new St.Icon({ gicon: gicon,
+                                 icon_size: iconSize });
+        icon.set_pivot_point(0.5, 0.5);
```

Comment:
I think you can pass the pivot point to the `St.Icon` constructor, like this:

```javascript
let icon = new St.Icon({
    gicon: gicon,
    icon_size: iconSize,
    pivot_point: new Clutter.Point({ x: 0.5, y: 0.5 }),
});
```

Reply:
Yes, that works.